src/validation/validator.js

```diff
@@ -10,13 +11,28 @@ const validate = (schema, property) => (req, res, next) => {
   const { error } = Joi.validate(req[property], schema);
   const valid = error == null;
 
-  if (valid) {
-    next();
-  } else {
-    return res.status(400).json({
-      error: error.details[0].message,
-    });
+  if (!valid) return Response.badRequest(res, error.details[0].message);
+
+  next();
+};
+
+/**
+  * Input validation logic
+  * @param {object} schema - schema to be used for input validation
+  * @param {string} property - property to be validated
+  * @return {json} - validation error
+  */
+export const validator = (schema, property = 'body') => (req, res, next) => {
+  const errors = Joi.validate(req[property], schema, { abortEarly: false });
+
+  if (errors.error) {
+    const errorMSG = [];
+
+    errors.error.details.forEach(err => errorMSG.push(err.message));
+
+    return Response.badRequest(res, errorMSG);
   }
+  next();
 };
 
```

Comment:
Please, why do we have two validators function in this file? the one @cleave3 created and yours.

Reply:
The first returns just single validation error while the one I create return more than one validation error for now I don't want others code to break that why I didn't touch the one created by @cleave3

Reply:
@SamuelKoroh , we should use one. you can adjust other validations to use the one that has multiple validation. and test login and signup to ensure nothing is breaking

Reply:
I did that and a lot of test were failing that's why I option to do  it separately.